advanced_alchemy/base.py

```diff
@@ -92,26 +95,26 @@ def merge_table_arguments(
     Args:
         cls (DeclarativeBase): This is the model that will get the table args
         *mixins (Any): The mixins to add into the model
-        table_args: additional information to add to tableargs
+        table_args: additional information to add to table_args
 
     Returns:
         tuple | dict: The merged __table_args__ property
     """
     args: list[Any] = []
     kwargs: dict[str, Any] = {}
 
-    mixin_table_args = (getattr(super(base_cls, cls), "__table_args__", None) for base_cls in (cls, *mixins))
+    mixin_table_args = (getattr(super(base_cls, cls), "__table_args__", None) for base_cls in (cls, *mixins))  # pyright: ignore[reportArgumentType,reportUnknownArgumentType]
 
     for arg_to_merge in (*mixin_table_args, table_args):
         if arg_to_merge:
             if isinstance(arg_to_merge, tuple):
                 last_positional_arg = arg_to_merge[-1]
                 args.extend(arg_to_merge[:-1])
                 if isinstance(last_positional_arg, dict):
-                    kwargs.update(last_positional_arg)
+                    kwargs.update(last_positional_arg)  # pyright: ignore[reportUnknownArgumentType]
                 else:
                     args.append(last_positional_arg)
-            elif isinstance(arg_to_merge, dict):
+            else:
```

Comment:
I believe this can only be a `dict` at this point, if `arg_to_merge` can be `None`, a `tuple` or a `dict`, then `dict` is all that's left.